Team04/dev - Bulk solution from team4 - Dynamic clock

johnnylazoq opened the pull request:
All changes:
New dynamic clock
Modularized vue components.
Fetching data from data team.
Different levels

DavidKarlJohnson committed:
Refactor CSS styling to comform to game-frame size restriciton

AndreasLyhagen committed:
testing sync

AndreasLyhagen committed:
Revert "testing sync"

This reverts commit 866c70f2a65b1122eb15897fdc911e3e39695031.

johnnylazoq committed:
Merge pull request #3 from uu-group4/team04/51-index-refactor

Refactor CSS styling to comform to game-frame size restriciton

johnnylazoq committed:
Merge branch 'uu-semp:main' into team04/dev

johnnylazoq committed:
Merge branch 'uu-semp:main' into main

DavidKarlJohnson committed:
Refactor CSS styling AGAIN to conform to game-frame

DavidKarlJohnson committed:
Remove game-frame.css link from index.html

Removed unused game-frame.css stylesheet link.

AndreasLyhagen committed:
Merge branch 'team04/dev' of https://github.com/uu-group4/swedish-learning-app-2025_new into team04/dev

lazjo-cgi committed:
Team04#58:created link between words, global assets and picture pointed to team04

ElysiaLiu1230 committed:
Team04#61: creat a start page

ElysiaLiu1230 committed:
Team04#55: Remove *header* and *footer* components from the *app* container

ElysiaLiu1230 committed:
Merge pull request #4 from uu-group4/team04/53-new-gameframe-adjust-elements

Refactor CSS styling AGAIN to conform to game-frame

DavidKarlJohnson committed:
Create boilerplate for dynamic clock

* Added clock image without hands as an asset.

* Created a clock-group, where all clock elements should be contained.

* Set clock image as background to clock-group.

* Added hour- and minute hands with simple divs

DavidKarlJohnson committed:
Position hour and minute hands in center of clock

* The hour and minute hands are now anchored to the clock.

* Length of the hour and minute hands are set with a static pixel size
so that they're within the borders of the clock.

* Hands of the clock have rounted corners for now, not pointed.

DavidKarlJohnson committed:
Create function for setting the clock

* 'setClock' function has been implemented, which will set the hour and
minute hands of the clock based on its arguments.

* This module has been added to a new folder called 'utils'.

* The clock is now set from 'app.js', which now has 'mounted()', and
will set the clock as render.

DavidKarlJohnson committed:
Add error handling to setClock function for invalid time arguments

DavidKarlJohnson committed:
Create documentation for 'setClock' JS function

ElysiaLiu1230 committed:
Merge branch 'team04/dev' of https://github.com/uu-group4/swedish-learning-app-2025_new into create-a-start-page

ElysiaLiu1230 committed:
#61:Merge pull request #5 from uu-group4/create-a-start-page

I solved the conflict, but I'm not sure if I did correctly. Please check it. Thanks a lot!

ElysiaLiu1230 committed:
merged and modify footer and heaader

ElysiaLiu1230 committed:
modify app.js to remove footer and header

ElysiaLiu1230 committed:
# 55: Merge pull request #6 from uu-group4/remove-footer-and-header

Remove footer and header

johnnylazoq committed:
Merge branch 'uu-semp:main' into main

lazjo-cgi committed:
readme was updated

johnnylazoq committed:
Merge branch 'uu-semp:main' into main

johnnylazoq committed:
Merge branch 'team04/dev' into team04/76-dynamic-clock

johnnylazoq committed:
Merge pull request #7 from uu-group4/team04/76-dynamic-clock

Create dynamic clock

lazjo-cgi committed:
test

lazjo-cgi committed:
testing

lazjo-cgi committed:
testing2

lazjo-cgi committed:
testing3

lazjo-cgi committed:
testing3

lazjo-cgi committed:
testing5

lazjo-cgi committed:
testing6

lazjo-cgi committed:
testing7

lazjo-cgi committed:
testing8

ElysiaLiu1230 committed:
Add start page component (clean version)

ElysiaLiu1230 committed:
#61: Add start page component

lazjo-cgi committed:
testing8

ElysiaLiu1230 committed:
Merge branch 'create-start-page-clean' of https://github.com/uu-group4/swedish-learning-app-2025_new into create-start-page-clean

ElysiaLiu1230 committed:
#61 create start page and modify app.js to remove unused components

lazjo-cgi committed:
Merge branch 'main' of github.com:uu-semp/swedish-learning-app-2025

lazjo-cgi committed:
Merge branch 'main', remote-tracking branch 'origin' into team04/dev

johnnylazoq committed:
Merge pull request #9 from uu-group4/create-start-page-clean

Team 04#61：Create start page clean & #57: Remove footer and header

lazjo-cgi committed:
corrected the word document to original pattern

johnnylazoq committed:
Merge branch 'uu-semp:main' into team04/53-new-gameframe-adjust-elements

johnnylazoq committed:
Merge branch 'Team04/dev' into team04/53-new-gameframe-adjust-elements

johnnylazoq committed:
Merge pull request #10 from uu-group4/team04/53-new-gameframe-adjust-elements

 feat: Adapt game layout to new static frame

johnnylazoq committed:
Merge branch 'uu-semp:main' into team04/76-dynamic-clock

johnnylazoq committed:
Merge branch 'Team04/dev' into team04/76-dynamic-clock

johnnylazoq committed:
Merge pull request #11 from uu-group4/team04/76-dynamic-clock

feat: Implement dynamic clock with setClock function

lazjo-cgi committed:
taken .csv from main

ElysiaLiu1230 committed:
#61: create start page and #57: remove header and footer

johnnylazoq committed:
Merge pull request #12 from uu-group4/#61--create-start-page-and-#57--remove-header-and-footer

Team04: #61: create start page and #57: remove header and footer

ElysiaLiu1230 committed:
Team04#81: Implement dynamic level loading, shuffled question handling, and interactive game logic for the Clock it! game.

lazjo-cgi committed:
Merge branch 'main' into Team04/dev

lazjo-cgi committed:
Merge branch 'Team04/dev' into #81---Implement-dynamic-level-loading-and-interactive-game-logic-for-game-

johnnylazoq committed:
Merge pull request #13 from uu-group4/#81---Implement-dynamic-level-loading-and-interactive-game-logic-for-game-

Team04#81: Implement dynamic level loading and finish playing logic in game

DavidKarlJohnson committed:
Refactor 'clock' component to controll clock hands

* Clock component is now controlled from its parent with props passed
down to it.

* Hour and minute hand functions (computed) added for changing the CSS
styling of the divs.

DavidKarlJohnson committed:
'gameView' no longer calls util function 'setClock'

* The 'gameView' component no longer calls on the helper function 'setClock'
in the utils folder. The 'gameView' now passes props to the 'clock' component
instead to set the clock.

DavidKarlJohnson committed:
Delete 'setClock' helper function, remove 'mounted' from 'app.js'.

* The 'setClock' helper function that was in the utils folder had been
removed.

* 'mount()' which had a 'setClock' in it has been removed from the
'app.js' file.

DavidKarlJohnson committed:
Team04#80: Add animation to clock hands

* Transition effect added to clock hands.

* Moved some CSS attributes from the minute- and hour hand selectors to
the 'clock-hand' selector.

DavidKarlJohnson committed:
Team04#80: Add shadow effect to clock hands

AndreasLyhagen committed:
Reset button now has functionality

AndreasLyhagen committed:
Merge pull request #14 from uu-group4/team04/78-refactor-dynamic-clock-vue

Team04/78 Refactor dynamic clock to use Vue instead of helper function

santabeard committed:
Team04#18user_progress

lazjo-cgi committed:
Merge branch 'main' into Team04/dev

lazjo-cgi committed:
changed back vocabulary.js

johnnylazoq committed:
Merge branch 'uu-semp:main' into Team04/dev

xFrednet reviewed (approved):
Changes to my files LGTM

ElysiaLiu1230 committed:
#56 Move or remove the 'Switch to FinishView' button

ElysiaLiu1230 committed:
#92: Redo statistics bar to give user instructions instead

DavidKarlJohnson committed:
Team04#89: Remove colored backgrounds

CSS background-color has been removed from individual components.
All backgrounds are in white, like the game-frame itself.

lazjo-cgi committed:
Merge branch 'main' into Team04/dev

xFrednet reviewed (approved):
My files LGTM

AndreasLyhagen committed:
Merge pull request #19 from uu-group4/team04/89-remove-colored-backgrounds

Team04#89: Remove colored backgrounds

johnnylazoq committed:
Merge pull request #17 from uu-group4/team04/80-animate-clock-hands

Team04/80 Animate clock hands

johnnylazoq committed:
Merge pull request #16 from uu-group4/#56--Move-or-remove-the-'Switch-to-FinishView'-button

Team04 #56: Move or remove the 'Switch to FinishView' button

johnnylazoq committed:
Merge pull request #18 from uu-group4/#92-Redo-statistics-bar-to-give-user-instructions-instead

#92: Redo statistics bar to give user instructions instead

DavidKarlJohnson committed:
Team04#87: Add hover animation to cards

Cards now have a box-shadow at all times.
When the user hovers over the button it'll increase the box-shadow and
scale the button.

DavidKarlJohnson committed:
Team04#87: Add ease effect to color change of cards

The background-color of the 'Card' buttons will now have an ease effect
so that the color changes slowly when transitioning to red/green. The
buttons also have an animation now when the user clicks.

DavidKarlJohnson committed:
Team04#87: Add shadowed text to 'Card' hover effect

DavidKarlJohnson committed:
Change styling of level select buttons. Move buttons within the navigation grouping

DavidKarlJohnson committed:
Team04#101: Organize all nav buttons in a row

All navigation buttons are now grouped together at the bottom, in a row,
of the game frame. All buttons are equal in size.

DavidKarlJohnson committed:
Team04#101: Create one unified nav bar

Only the buttons furthes to the left and right have rounded corners.
All others have no rounded corners. Therefore, the whole nav will look
like one unified button.

DavidKarlJohnson committed:
Team04#101: Add hover effects to buttons

DavidKarlJohnson committed:
Team04#101: Create "blinking" animation for next button

The 'Next question' button will now blink in green when the user
has made a guess. Indicating where they should click to move forwards.

santabeard committed:
team04#94stores answers in the session

santabeard committed:
team04#94going forward progress save

santabeard committed:
team04#18modularized_CSS

santabeard committed:
Merge branch '#94-keep-answers-when-going-backwards' into #18-session-progress-without-correctness-indication

santabeard committed:
Revert "Merge branch '#94-keep-answers-when-going-backwards' into #18-session-progress-without-correctness-indication"

This reverts commit 265da61de9519a24c1c82e9a4c0e8bf39cdc38b3, reversing
changes made to b049ed19dcefc17736324ced42f78cc305ad2200.

UrWoo committed:
Merge pull request #21 from uu-group4/team04/101-change-navigation-styling

Team04/101: Update 'Navigation' CSS styling

santabeard committed:
Update gameView.js

lundandreas committed:
Finish page with stats

Added support for tracking statistics of a game session which is displayed on the finish page. Also some CSS for the finish

johnnylazoq committed:
Merge pull request #22 from uu-group4/#18-session-progress-without-correctness-indication

#18 session progress without correctness indication

lazjo-cgi committed:
Merge branch 'main' into Team04/dev

johnnylazoq committed:
Merge pull request #20 from uu-group4/team04/87-change-cards-styling

Team04/87: Animate and change 'Cards' styling

lundandreas committed:
Merge branch 'Team04/dev' into Team04/-#30-create-game-finish-page

DavidKarlJohnson committed:
Team04#103: Add thicker border to correct question

When the user makes their selection - the border around the correct 'Card'
will be highlighted. This is to add accessibility for color blind users.

johnnylazoq committed:
Merge pull request #23 from uu-group4/Team04/-#30-create-game-finish-page

Finish page with stats

johnnylazoq committed:
Merge pull request #25 from uu-group4/team04/103-add-border-to-correct-card

Team04#103: Add thicker border to correct question

DavidKarlJohnson committed:
Merge branch 'Team04/dev' into team04/91-adding-functionality-to-reset-button

johnnylazoq committed:
Merge pull request #15 from uu-group4/team04/91-adding-functionality-to-reset-button

Reset button now has functionality

santabeard committed:
Merge branch 'Team04/dev' into Team04#94-keeping-user-answers-when-going-forward

santabeard committed:
team04#94 keeps user answers when stepping through questions

johnnylazoq committed:
Merge pull request #26 from uu-group4/Team04#94-keeping-user-answers-when-going-forward

Team04#94 keeping user answers when going forward

lazjo-cgi committed:
Merge branch 'main' into Team04/dev

santabeard committed:
Merge branch 'Team04/dev' into #93-locking-cards-after-selection

santabeard committed:
team04#93 fixed the naming

johnnylazoq committed:
Merge pull request #27 from uu-group4/#93-locking-cards-after-selection

Update gameView.js for locking answers

lundandreas committed:
New license safe clock image

lundandreas committed:
team04#21 updated clock hand width

updated clock hand width to match new clock image

DavidKarlJohnson committed:
Team04#106: Fix non-blinking next button

DavidKarlJohnson committed:
Remove console.log from 'highlightNextQuesionBtn'

DavidKarlJohnson committed:
Team04#107: Change startView to have welcome message

The start view now features a welcome message instead of instructions
for the different levels.

ElysiaLiu1230 committed:
Merge pull request #28 from uu-group4/Team04/21-create-images-for-the-clock

Team04/21 create images for the clock

lazjo-cgi committed:
Merge branch 'main' into Team04/dev

lazjo-cgi committed:
Merge branch 'Team04/dev' of github.com:uu-group4/swedish-learning-app-2025_new into Team04/dev

johnnylazoq committed:
Merge pull request #29 from uu-group4/team04/106-fix-non-blinking-next-btn

Team04/106 Fix non blinking next btn

johnnylazoq committed:
Merge pull request #30 from uu-group4/team04/107-change-game-instructions

Team04/107 Change game instructions

lazjo-cgi committed:
Merge branch 'main' into Team04/dev
